quiche/src/recovery/bbr/mod.rs

```diff
@@ -310,9 +308,14 @@ fn on_packets_acked(
             acked_bytes + p.size
         });
 
+    // Upon entering Fast Recovery.
+    if !r.bbr_state.in_recovery && r.congestion_recovery_start_time.is_some() {
+        bbr_enter_recovery(r);
+    }
+
     if let Some(pkt) = packets.last() {
-        if !r.in_congestion_recovery(pkt.time_sent) && r.bbr_state.in_recovery {
-            // Upon exiting loss recovery.
+        // Upon exiting loss recovery.
+        if !r.in_congestion_recovery(pkt.time_sent) {
```

Comment:
This check leads to calling bbr_exit_recovery even though we are not in recovery!

Reply:
I'd like to understand which case can create such condition. in_congestion_recovery() checks should return `false` during non-recovery phase.

Reply:
Thought it again and it's correct.

Reply:
Isn't we should call bbr_exit_recovery() only if we are in recovery phase? If so then above code will call bbr_exit_recovery() even  though you are not in recovery.

Reply:
Oh, I mean I updated the code as you proposed originally. :)